controllers/iperf3/clientpod.go

```diff
@@ -17,28 +17,27 @@ limitations under the License.
 package iperf3
 
 import (
-	"context"
 	"fmt"
 	"strings"
 
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
-
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
-	"github.com/xridge/kubestone/pkg/k8s"
-
 	perfv1alpha1 "github.com/xridge/kubestone/api/v1alpha1"
 )
 
 // +kubebuilder:rbac:groups="",resources=pods,verbs=get;list;create;delete
 
-func (r *Reconciler) newClientPod(ctx context.Context, cr *perfv1alpha1.Iperf3, crRef *corev1.ObjectReference) error {
+func podName(cr *perfv1alpha1.Iperf3) string {
+	return cr.Name + "-client"
+}
+
+func newClientPod(cr *perfv1alpha1.Iperf3) metav1.Object {
```

Comment:
as the `newClientPod`, `newServerDeployment` and `newServerService` is simplified we could create unit tests to check if all the values from the CR are correctly populated.

Reply:
I took it one step further: after unit testing it is exposed it as public api.